File: Generation/DSA01017.cpp
#include <bits/stdc++.h>
using namespace std;

#define tests()     int test_cases; cin >> test_cases; while (test_cases--)

string binaryToGray(const string& binary) {
    string gray = "";
    gray += binary[0]; 
    for (size_t i = 1; i < binary.size(); i++) {
        gray += (binary[i] - '0') ^ (binary[i - 1] - '0') + '0';
    }
    return gray;
}

int main() {
    tests() {
        string binary;
        cin >> binary;
        cout << binaryToGray(binary) << endl;
    }
    return 0;
}

File: Java/Generation/DSA01017.java
import java.util.Scanner;

public class DSA01017 {
    public static String binaryToGray(String binary) {
        StringBuilder gray = new StringBuilder();
        gray.append(binary.charAt(0));
        
        for (int i = 1; i < binary.length(); ++i) {
            char grayBit = (char) (((binary.charAt(i) - '0') ^ (binary.charAt(i - 1) - '0')) + '0');
            gray.append(grayBit);
        }
        return gray.toString();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int testCases = scanner.nextInt();
        while (testCases-- > 0) {
            String binary = scanner.next();
            System.out.println(binaryToGray(binary));
        }
        scanner.close();
    }
}
